packages/flutter_tools/lib/src/ios/ios_workflow.dart

```diff
@@ -125,66 +125,66 @@ class IOSValidator extends DoctorValidator {
-        messages.add(ValidationMessage.error(
-            'libimobiledevice and ideviceinstaller are not installed. To install, run:\n'
-            '  brew install --HEAD libimobiledevice\n'
-            '  brew install ideviceinstaller'
-        ));
-      } else if (!await iMobileDevice.isWorking) {
-        brewStatus = ValidationType.partial;
+    final bool iHasIosDeploy = await hasIosDeploy;
+
+    // Check ios-deploy is installed at meets version requirements.
+    if (iHasIosDeploy) {
+      messages.add(
+        ValidationMessage('ios-deploy ${await iosDeployVersionText}'));
+    }
+    if (!await _iosDeployIsInstalledAndMeetsVersionCheck) {
+      packageManagerStatus = ValidationType.partial;
+      if (iHasIosDeploy) {
         messages.add(ValidationMessage.error(
-            'Verify that all connected devices have been paired with this computer in Xcode.\n'
-            'If all devices have been paired, libimobiledevice and ideviceinstaller may require updating.\n'
-            'To update, run:\n'
-            '  brew uninstall --ignore-dependencies libimobiledevice\n'
-            '  brew install --HEAD libimobiledevice\n'
-            '  brew install ideviceinstaller'
+          'ios-deploy out of date ($iosDeployMinimumVersion is required). To upgrade with Brew:\n'
+          '  brew upgrade ios-deploy'
         ));
-      } else if (!await hasIDeviceInstaller) {
-        brewStatus = ValidationType.partial;
+      } else {
         messages.add(ValidationMessage.error(
-          'ideviceinstaller is not installed; this is used to discover connected iOS devices.\n'
-          'To install, run:\n'
-          '  brew install --HEAD libimobiledevice\n'
-          '  brew install ideviceinstaller'
+          'ios-deploy not installed. To install with Brew:\n'
+          '  brew install ios-deploy'
         ));
       }
-
-      // Check ios-deploy is installed at meets version requirements.
-      if (await hasIosDeploy) {
-        messages.add(ValidationMessage('ios-deploy ${await iosDeployVersionText}'));
-      }
-      if (!await _iosDeployIsInstalledAndMeetsVersionCheck) {
-        brewStatus = ValidationType.partial;
-        if (await hasIosDeploy) {
-          messages.add(ValidationMessage.error(
-            'ios-deploy out of date ($iosDeployMinimumVersion is required). To upgrade:\n'
-            '  brew upgrade ios-deploy'
-          ));
-        } else {
-          messages.add(ValidationMessage.error(
-            'ios-deploy not installed. To install:\n'
-            '  brew install ios-deploy'
-          ));
-        }
-      }
-
-    } else {
-      brewStatus = ValidationType.missing;
+    }
+    // If one of the checks for the packages failed, we need brew so that we can install
+    // the necessary packages. If they're all there, however, we don't even need brew.
+    if (packageManagerStatus == ValidationType.partial && !hasHomebrew) {
+      packageManagerStatus = ValidationType.missing;
```

Comment:
Shouldn't this still be a partial status?

Reply:
Unless they're all missing? But yah, I guess it could be missing or partial here.